site/README.md

```diff
@@ -0,0 +1,182 @@
+
+### MkDocs background 
+
+In MkDocs, the [`docs_dir`](https://www.mkdocs.org/user-guide/configuration/#docs_dir) points to the root directory containing the source markdown files for an MkDocs project. By default, this points to the `docs` directory. When you build MkDocs `mkdocs build`, MkDocs generates the static site in the [`site_dir`](https://www.mkdocs.org/user-guide/configuration/#site_dir) becomes the root of that project for the generated site. 
+
+### Iceberg docs layout
+
+In the Iceberg docs, since the top-level site and versioned docs are contained in the same directory, they all live under the `/site` directory of the main Iceberg repository. The `/site/docs` directory is named this way to follow the [MkDocs convention](https://www.mkdocs.org/user-guide/configuration/#docs_dir), while the `/site/docs/docs` directory is an analog to the "Docs" navigation tab. Under this directory, you'll find the `/site/docs/docs/nightly` directory, which contains the state of the documentation in the local revisions.
+
+The non-versioned site pages are all the `/site/docs/.*md` files and the docs are the `/site/docs/docs/<version>/docs/*.md` files. Notice the location of the `mkdocs.yml`. Looking at this though, you may ask where the older versions and javadocs are.
+
+```
+./site/
+├── docs
+│   ├── assets
+│   ├── docs
+│   │   └── nightly
+│   │       ├── docs
+│   │       │   ├── assets
+│   │       │   ├── api.md
+│   │       │   ├── ...
+│   │       │   └── table-migration.md
+│   │       └── mkdocs.yml (versioned)
+│   ├── about.md
+│   ├── ...
+│   └── view-spec.md
+├── README.md
+├── mkdocs.yml (non-versioned)
+├── requirements.txt
+└── variables.yml
+```
+### Building the versioned docs
+
+All previously versioned docs will be committed in `docs-<version>` branches and mounted using [git worktree](https://git-scm.com/docs/git-worktree) at build time. The worktree will pull these versions in following the `/site/docs/docs/<version>` convention. The `latest` version, will be a secondary copy of the most recent build version in the worktree, but pointing to `/site/docs/docs/latest`. There is also a `javadoc` branch that contains all prior static generation versions of the javadocs in a single tag.
+
+```
+./site/
+└── docs
+    ├── docs
+    │   ├── nightly
+    │   ├── latest
+    │   ├── 1.3.1
+    │   ├── 1.3.0
+    │   └── ...
+    └── javadoc
+        ├── latest
+        ├── 1.3.1
+        ├── 1.3.0
+        └── ...
+```
+
+### Install
+
+1. (Optional) Set up venv
+```
+python -m venv mkdocs_env
+source mkdocs_env/bin/activate
+```
+
+1. Install required Python libraries
+```
+pip install -r requirements.txt
+```
+
+#### Adding additional versioned documentation
+
+To build locally with additional docs versions, add them to your working tree.
+For now, I'm just adding a single version, and the javadocs directory.
+
+```
+git worktree add site/docs/docs/1.3.1 docs-1.3.1
+git worktree add site/docs/javadoc javadoc
```

Comment:
These don't exist yet, so I had to comment out the 1.3.1 entry in mkdocs.yml to get `mkdocs serve` working. Not a blocker.

Reply:
Added a warning to the README